projects/ion/src/lib/dropdown/dropdown.component.html

```diff
@@ -33,4 +33,15 @@
       ></ion-icon>
     </div>
   </div>
+  <ion-button
+    label="Limpar"
+    iconType="trash"
+    type="secondary"
+    size="sm"
+    rightSideIcon="true"
+    expand="true"
+    *ngIf="clearFiltersSelected()"
```

Comment:
Acredito que em uma verificação condicional, checar um valor de uma variável seja mais fácil e legível do que analisar o retorno de uma função. Minha sugestão é trocar ```*ngIf="clearFiltersSelected()"``` por ```*ngIf="clearButtonIsVisible"```. 
Pode ser outro nome, viu 😄  

Na função ```clearOption() { ... }``` você colocaria a visualização para falso ```this.clearButtonIsVisible = false```.

E na função ```select(option: DropdownItem) { ... }``` chamaria a função ```setVisibleClearButton()``` na qual essa função definiria a visualização do botão. 

```
setVisibleCleanButton() {
    this.clearButtonIsVisible = this.options.some(
      (option) => option.selected == true
    );
}
```
O método some testa se pelo menos um elemento dentro do array (no caso, options) corresponde a validação passada para ela ```  (option) => option.selected == true ```. Se pelo menos uma elemento for verdadeiro, ela retornará true, caso contrário, retornará falso.